### PR TITLE
[AIR] Add `SklearnCheckpoint`

### PR DESCRIPTION
**Pull request by bveeramani**
## Why are these changes needed?

These changes are needed to make checkpoint operations more discoverable. For context, see https://github.com/ray-project/ray/pull/26651.

## Related issue number

https://github.com/ray-project/ray/pull/26651

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [x] This PR is not tested :(

### CODE REVIEW

File: python/ray/air/checkpoint.py [hunk 1]
```diff
@@ -351,6 +351,15 @@ def from_directory(cls, path: str) -> "Checkpoint":
         """
         return cls(local_path=path)
 
+    @classmethod
+    def _from_checkpoint(cls, other: "Checkpoint") -> "Checkpoint":
```
Comment (ericl) [inline python/ray/air/checkpoint.py:355]: Should we port the others to use this one?

Comment (ericl) [inline python/ray/air/checkpoint.py:355]:
```suggestion
    def copy_from(cls, other: "Checkpoint") -> "Checkpoint":
```

**Review by ericl — APPROVED** [review]

File: python/ray/air/checkpoint.py (hunk 1 above)
Comment (bveeramani) [reply python/ray/air/checkpoint.py:355]: I will

Comment (bveeramani) [reply python/ray/air/checkpoint.py:355]: Could we make the method private? I'm not sure whether we should expose this functionality to the user.

```suggestion
    def _copy_from(cls, other: "Checkpoint") -> "Checkpoint":
```

Comment (ericl) [reply python/ray/air/checkpoint.py:355]: We need to expose the cast operation to the user. So either we expose `copy_from()`, or we expose it in the constructor of the framework-specific checkpoint (we should pick one or the other).

**Comment by bveeramani** [comment]
Closed in favor of https://github.com/ray-project/ray/pull/26777

**Pull request closed without merging**